specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-11-20-preview/hanaonazure.json

```diff
@@ -0,0 +1,880 @@
+            "schema": {
+              "$ref": "#/definitions/SapMonitor"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "HanaOnAzure"
+        ],
+        "operationId": "SapMonitors_Delete",
+        "x-ms-long-running-operation": true,
+        "summary": "Deletes a SAP monitor.",
+        "description": "Deletes a SAP monitor with the specified subscription, resource group, and monitor name.",
+        "x-ms-examples": {
+          "Deletes a SAP monitor": {
+            "$ref": "./examples/SapMonitors_Delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SapMonitorNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Deleted SapMonitor"
+          },
+          "202": {
+            "description": "Deleting SapMonitor"
+          },
+          "204": {
+            "description": "No SapMonitor to delete"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "HanaOnAzure"
+        ],
+        "operationId": "SapMonitors_Update",
+        "summary": "Patches the Tags or identity field of a SAP monitor.",
+        "description": "Patches the Tags or identity field of a SAP monitor for the specified subscription, resource group, and monitor name.",
+        "x-ms-examples": {
+          "Update Tags field of a SAP monitor": {
+            "$ref": "./examples/SapMonitors_PatchTags.json"
+          },
+          "Delete Tags field of a SAP monitor": {
+            "$ref": "./examples/SapMonitors_PatchTags_Delete.json"
+          },
+          "Update Identity field of a SAP monitor": {
+            "$ref": "./examples/SapMonitors_PatchIdentity.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SapMonitorNameParameter"
+          },
+          {
+            "$ref": "#/parameters/TagsParameter"
+          },
+          {
+            "$ref": "#/parameters/IdentityParameter"
```

Comment:
>IdentityParameter [](start = 34, length = 17)

This is not the correct way to model a managed identity on a patch request because the parameter is being passed through the path, which is not allowed in ARM.

The current `TagsParameter` is marked as a `body` parameter. `TagsParameter` needs to be modifed so the identity and tags come in through the request body.

Reply:
The latest update makes `TagParameter` and `IdentityParameter` both be body parameters. Assuming this would even work in autorest, the generated SDK will be unnatural. You should combine TagsParameter and IdentityParameter into one. Take a look at Microsoft.Compute/virtualMachines resources in swagger as an example.

---
In reply to: [530054875](https://github.com/Azure/azure-rest-api-specs/pull/11796#discussion_r530054875) [](ancestors = 530054875)

Reply:
Thanks. Will do.

Reply:
I have combined the two into one. Please review @majastrz

Reply:
Please check the comments I left. I still do not see the TagsIdentityParameter in patch definition.